src/include/cell.i

```diff
@@ -706,19 +710,32 @@ __wt_cell_leaf_value_parse(WT_PAGE *page, WT_CELL *cell)
     }
 }
 
+/*
+ * The verification code specifies an end argument, a pointer to 1B past the end-of-page. In which
+ * case, make sure all reads are inside the page image. If an error occurs, return an error code but
+ * don't output messages, our caller handles that.
+ */
+#define WT_CELL_LEN_CHK(t, len)                                                             \
+    do {                                                                                    \
+        if (end != NULL &&                                                                  \
+          ((uint8_t *)(t) < (uint8_t *)dsk || (((uint8_t *)(t)) + (len)) > (uint8_t *)end)) \
+            return (WT_ERROR);                                                              \
+    } while (0)
+
 /*
  * __wt_cell_unpack_safe --
  *     Unpack a WT_CELL into a structure, with optional boundary checks.
  */
 static inline int
 __wt_cell_unpack_safe(WT_SESSION_IMPL *session, const WT_PAGE_HEADER *dsk, WT_CELL *cell,
-  WT_CELL_UNPACK *unpack, const void *end)
+  WT_CELL_UNPACK_ADDR *unpack_addr, WT_CELL_UNPACK_KV *unpack_value, const void *end)
```

Comment:
Can we break this into two functions? We may be accidentally passing both `unpack_addr` and `unpack_value` to this. To share common code between the two functions, we can share them through small inline functions.

Reply:
No. There is a single function that cracks a cell, and now that we crack cells into different structures, this function has to take references to both types. One of the arguments is always NULL, that is, we're always working with one type or another.

Obviously, we could split this function into two parts, but that's not something I'm eager do.